modules/apps/content-dashboard/content-dashboard-api/src/main/java/com/liferay/content/dashboard/item/action/ContentDashboardItemActionProvider.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.dashboard.item.action;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author David Arques
+ */
+public interface ContentDashboardItemActionProvider<T> {
+
+	public ContentDashboardItemAction getContentDashboardItemAction(
```

Comment:
@cgoncas I'm still not sure if this should return an Optional

Reply:
I think it is really strange to register and action provider that returns empty actions.